3D/senior-csharp-test/Assets/Cache/FileCacheStorage.cs

```diff
@@ -269,4 +184,86 @@ private void UpdateDataEntries()
 			}
 		}
 	}
+
+	public byte[] Get( string id )
+	{
+		if( !_cacheEntries.TryGetValue( id, out FileCacheEntry ce ) )
+			return null;
+
+		UpdateUseDate( id );
+		return _fileSystem.Read( ce.Data );
+	}
+
+	public bool Has( string id )
+    {
+        return _cacheEntries.ContainsKey( id );
+    }
+
+	public bool MatchesVersion( string id, string version )
+	{
+		try
+		{
+			if( _cacheEntries[ id ].Version == version )
+				return true;
+		}
+		catch( Exception e )
+		{
+			Log.LogError( "Cache entry id = " + id + " not found : " + e );
+		}
+
+		return false;
+	}
+
+	public void Set( byte[] value, string id, string version = null )
+	{
+		string path = _cachePath + "/" + id;
+
+		if( !Has( id ) || !_fileSystem.FileExists( path ) || !MatchesVersion( id, version ) )
+			SaveFile( path, value );
+
+		_cacheEntries[ id ] = new FileCacheEntry( id, version, path, DateTime.Now.ToString( DATE_FORMAT ) );
+	}
+
+	public void Delete( string id )
+	{
+		if( !_cacheEntries.TryGetValue( id, out FileCacheEntry ce ) )
+			return;
+
+		try
+		{
+			DeleteFile( ce );
+		}
+		catch( Exception )
+		{
+			Log.LogError( "Fail to delete file : " + ce );
+			return;
+		}
+
+		_cacheEntries.Remove( id );
```

Comment:
`_cacheEntries.Remove( id )` will not be executed if there is an exception in `DeleteFile( ce )`

Reply:
Isn't it fine to let the entry in the dictionary if we fail to delete and the assetbundle file still exist?

Reply:
ok